.changelog/13606.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+api: refactor ACL check when using the all namespaces wildcard
+```
```

Comment:
If the external behavior doesn't change, we shouldn't provide a changelog. If the external behavior changes, we should describe that change here instead.